pkgs/tools/typesetting/tex/texlive/frozen.nix

```diff
@@ -0,0 +1,36 @@
+{ callPackage, lib, fetchurl, useFixedHashes ? true }:
+let
+  version = {
+    texliveYear = 2022;
+    final = true;
+  };
+
+  urlPrefixes = with version;
+     [
+      # tlnet-final snapshot; used when texlive.tlpdb is frozen
+      # the TeX Live yearly freeze typically happens in mid-March
+      "http://ftp.math.utah.edu/pub/tex/historic/systems/texlive/${toString texliveYear}/tlnet-final"
+      "ftp://tug.org/texlive/historic/${toString texliveYear}/tlnet-final"
+    ];
+
+  tlpdbxz = fetchurl {
+    urls = map (up: "${up}/tlpkg/texlive.tlpdb.xz") urlPrefixes;
```

Comment:
The URL structure of a TeX Live repository is always the same, so this computation can be shifted to `default.nix` (so this would just be the hash of `tlpdbxz`).

Reply:
it actually is slightly different in `latest.nix` as we explicitly select the last mirror there. But this also works for the frozen.version, so it could be factored out